include/openmc/distribution_spatial.h

```diff
@@ -136,6 +136,34 @@ class MeshSpatial : public SpatialDistribution {
                                 //!< mesh element indices
 };
 
+//==============================================================================
+//! Distribution of points
+//==============================================================================
+
+class PointCloud : public SpatialDistribution {
+public:
+  explicit PointCloud(pugi::xml_node node);
+  explicit PointCloud(gsl::span<const double> x, gsl::span<const double> y,
+    gsl::span<const double> z, gsl::span<const double> strengths);
+
+  //! Sample a position from the distribution
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled position
+  Position sample(uint64_t* seed) const override;
+
+  //! Sample a point
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled point index
+  int32_t sample_point_index(uint64_t* seed) const;
+
+  double total_strength() { return this->point_idx_dist_.integral(); }
```

Comment:
I didn't think through normalization.  I don't know/recall if a `DiscreteIndex` is normalized, but I guess that it isn't given that a method like this exists for the `MeshSpatial` class.

Reply:
It appears it is in the `init_alias` method, but the `integral` method retains the original, potentially un-normalized, sum of the probabilities.